### PR TITLE
defined RamdomEngine class

### PR DESCRIPTION
**Pull request by DivasGupta7**

### CODE REVIEW

**Comment by anjanik012** [comment]
This is technically a declaration and not definition.

File: src/engine.h [hunk 1]
```diff
@@ -1,4 +1,20 @@
-// Define this class here.
+//Define this class here.
+#include <stdint.h>
```
Comment (anjanik012) [inline src/engine.h]: There should be `cstdint` header file or integer definitions should work without including this. Test this out

File: src/engine.h [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+#include <iostream>
 class RandomEngine {
```
Comment (anjanik012) [inline src/engine.h:3]: Please reformat code. Indent spaces properly.

File: src/engine.h [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
-
+    private:
+    uint8_t length;
+    bool symb;
+    public:
+    //constructors:
+    RandomEngine();
+    RandomEngine(uint8_t l);
```
Comment (anjanik012) [inline src/engine.h]: constructors with exactly one argument should be marked `explicit`

File: src/engine.h [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+    RandomEngine(bool s);
+    RandomEngine(uint8_t l,bool s);
+    //member functions:
+    void setLength(uint8_t);
+    void setSymbol(bool);
+    std::string getString();
+    int getLength();
```
Comment (anjanik012) [inline src/engine.h]: return type should be `uint8_t`

**Review by anjanik012 — CHANGES_REQUESTED** [review]

**Comment by anjanik012** [comment]
Also, where are definitions, are you working on it?

**Comment by DivasGupta7** [comment]
> Also, where are definitions, are you working on it?

yes sir,we are working on it in engine.cpp file

File: src/engine.h [hunk 5]
```diff
@@ -1,4 +1,20 @@
...
+         RandomEngine();
+         RandomEngine(uint8_t l);
+         RandomEngine(bool s);
+         RandomEngine(uint8_t l,bool s);
 
+    // member functions:
+         void setLength(uint8_t);
+         void setSymbol(bool);
+         std::string getString();
+         int getLength();
```
Comment (anjanik012) [inline src/engine.h]: Return type is incorrect.

**Review by anjanik012 — CHANGES_REQUESTED** [review]

File: src/engine.h [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+         RandomEngine();
+         explicit RandomEngine(uint8_t l);
+         explicit RandomEngine(bool s);
+         RandomEngine(uint8_t l,bool s);
 
+    // member functions:
+         void setLength(uint8_t);
+         void setSymbol(bool);
+         std::string getString();
+         u_int8_t getLength();
```
Comment (anjanik012) [inline src/engine.h]: still incorrect, correct is `uint8_t`. Look what you commit!!

File: src/engine.h [hunk 7]
```diff
@@ -1,4 +1,20 @@
-// Define this class here.
+// Declare this class here
+#include<iostream>
 class RandomEngine {
+    private:
+         uint8_t length=0;
```
Comment (anjanik012) [inline src/engine.h]: This is not the way to initialize variables, all class members must be initialized via constructors

**Review by anjanik012 — CHANGES_REQUESTED** [review]

**Comment by anjanik012** [comment]
There are a few more changes that I would like to have but never mind. Merging.

**Pull request merged**